driver/src/test/java/org/neo4j/driver/internal/connector/socket/ChunkedInputTest.java

```diff
@@ -150,4 +152,30 @@ public void canReadAllIntegerSizes() throws Exception
         assertEquals( 0,                 in.readInt() );
         assertEquals( Integer.MIN_VALUE, in.readInt() );
     }
-}
\ No newline at end of file
+
+    @Test
+    public void shouldNotReadMessageEndingWhenByteLeftInBuffer()
```

Comment:
The experience of using `exception.expect` is not so good for me. Sometimes if we have many places that might throw an exception then the `exception.expect` could catch any of them. So I usually just use the simple `try-fail-catch-message` to make sure that the exception is the one I want and it really comes from the line I want.